test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/ProjectFactories.cs

```diff
@@ -0,0 +1,148 @@
+                projectServices,
+                threadingService);
+            return testProject;
+        }
+
+        internal static LegacyPackageReferenceProject CreateLegacyPackageReferenceProject(TestDirectory testDirectory, string projectId, string range, IVsProjectThreadingService threadingService)
+        {
+            var onedep = new[]
+            {
+                new LibraryDependency
+                {
+                    LibraryRange = new LibraryRange(
+                        "packageA",
+                        VersionRange.Parse(range),
+                        LibraryDependencyTarget.Package)
+                }
+            };
+
+            return CreateLegacyPackageReferenceProject(testDirectory, projectId, threadingService, onedep);
+        }
+
+        internal static IVsProjectAdapter CreateProjectAdapter(string fullPath)
+        {
+            var projectAdapter = CreateProjectAdapter();
+            projectAdapter
+                .Setup(x => x.FullProjectPath)
+                .Returns(Path.Combine(fullPath, "foo.csproj"));
+            projectAdapter
+                .Setup(x => x.GetTargetFrameworkAsync())
+                .ReturnsAsync(NuGetFramework.Parse("netstandard13"));
+
+            var testMSBuildProjectExtensionsPath = Path.Combine(fullPath, "obj");
+            Directory.CreateDirectory(testMSBuildProjectExtensionsPath);
+            projectAdapter
+                .Setup(x => x.GetMSBuildProjectExtensionsPathAsync())
+                .Returns(Task.FromResult(testMSBuildProjectExtensionsPath));
+
+            return projectAdapter.Object;
+        }
+
+        internal static Mock<IVsProjectAdapter> CreateProjectAdapter()
+        {
+            var projectAdapter = new Mock<IVsProjectAdapter>();
+
+            projectAdapter
+                .SetupGet(x => x.ProjectName)
+                .Returns("TestProject");
+
+            projectAdapter
+                .Setup(x => x.GetRuntimeIdentifiersAsync())
+                .ReturnsAsync(Enumerable.Empty<RuntimeDescription>);
+
+            projectAdapter
+                .Setup(x => x.GetRuntimeSupportsAsync())
+                .ReturnsAsync(Enumerable.Empty<CompatibilityProfile>);
+
+            projectAdapter
+                .Setup(x => x.Version)
+                .Returns("1.0.0");
+
+            return projectAdapter;
+        }
+
+        internal static ProjectNames GetTestProjectNames(string projectPath, string projectUniqueName)
+        {
+            var projectNames = new ProjectNames(
+            fullName: projectPath,
+            uniqueName: projectUniqueName,
+            shortName: projectUniqueName,
+            customUniqueName: projectUniqueName,
+            projectId: Guid.NewGuid().ToString());
+            return projectNames;
+        }
+
+        internal static PackageSpec GetPackageSpec(string projectName, string packageSpecFullPath, string version)
+        {
+            string referenceSpec = $@"
```

Comment:
ProjectTestHelpers already has an appropriate method. 

You can look into fixing it, or I can do it in a follow up PR. 

I recently touched these codepaths so I have a good idea how we can minimize the duplication.

Reply:
I use both versions: `ProjectFactories.GetPackageSpec()` and `ProjectTestHelpers.GetDGSpecFromPackageSpecs(packageSpec)`. Let's follow up in https://github.com/NuGet/Client.Engineering/issues/1441